numba/tests/test_unicode.py

```diff
@@ -1173,7 +1177,19 @@ def test_getitem(self):
         cfunc = njit(pyfunc)
 
         for s in UNICODE_EXAMPLES:
-            for i in range(-len(s)):
+            for i in range(-len(s), len(s)):
+                self.assertEqual(pyfunc(s, i),
+                                 cfunc(s, i),
+                                 "'%s'[%d]?" % (s, i))
+
+    def test_getitem_scalar_kind(self):
+        # See issue #6135, make sure that getitem returns a char of the minimal
+        # kind required to represent the "got" item.
```

Comment:
You may want to indicate how condition is ensured here. Also, for the ones below.

Reply:
hash does it? is that what you are implying?

Reply:
Yes, this is what I assumed, but maybe write "use ` hash` to ensure .... as it does X`?

Reply:
Done in bab2a17

Reply:
👌